src/quadtree.h

```diff
@@ -0,0 +1,65 @@
+#include "otpch.h"
+
+#include "creature.h"
+#include "spectators.h"
+
+namespace tfs::quadtree {
+void find(uint16_t start_x, uint16_t start_y, uint16_t end_x, uint16_t end_y, SpectatorVec& spectators,
+          std::function<bool(Creature*)> comparasion);
+Tile* find_tile(uint16_t x, uint16_t y, uint8_t z);
+void create_tile(uint16_t x, uint16_t y, uint8_t z, Tile* tile);
+void move_creature(uint16_t old_x, uint16_t old_y, uint8_t old_z, uint16_t x, uint16_t y, uint8_t z,
+                   Creature* creature);
+void insert_creature(uint16_t x, uint16_t y, uint8_t z, Creature* creature);
+void remove_creature(uint16_t x, uint16_t y, uint8_t z, Creature* creature);
+} // namespace tfs::quadtree
+
+class QuadTree
+{
+public:
+	explicit QuadTree() = default;
+
+	virtual bool is_leaf() = 0;
+
+	std::unique_ptr<QuadTree> nodes[4] = {};
+};
+
+static constexpr int32_t FLOOR_BITS = 3;
+static constexpr int32_t FLOOR_SIZE = (1 << FLOOR_BITS);
+static constexpr int32_t FLOOR_MASK = (FLOOR_SIZE - 1);
```

Comment:
```suggestion
inline constexpr int32_t FLOOR_BITS = 3;
inline constexpr int32_t FLOOR_SIZE = (1 << FLOOR_BITS);
inline constexpr int32_t FLOOR_MASK = (FLOOR_SIZE - 1);
```

`static` was a mistake here

Reply:
Thanks